File: solution/1_CountMatchesInString/1_CountMatchesInString/CountMatchesInStringTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using countmatches;

namespace test.countmatches
{
    [TestClass]
    public class CountMatchesInStringTest
    {
        [TestMethod]
	    public void noMatchesInNullString() {
		    Assert.AreEqual(0, StringUtil.countMatches(null, 'b'));
	    }
	
	    [TestMethod]
	    public void noMatchesInEmptyString() {
            Assert.AreEqual(0, StringUtil.countMatches("", 'b'));
	    }
	
	    [TestMethod]
	    public void noMatchesInNonEmptyString() {
            Assert.AreEqual(0, StringUtil.countMatches("deze string evat niet de letter ee", 'b'));
	    }
	
	    [TestMethod]
	    public void oneMatchInNonEmptyString() {
            Assert.AreEqual(1, StringUtil.countMatches("deze string bevat de letter ee", 'b'));
	    }

        [TestMethod]
	    public void multipleMatchesInNonEmptyString() {
            Assert.AreEqual(3, StringUtil.countMatches("deze string bevat wel de letter b, meer dan 1 keer de letter b zelfs", 'b'));
	    }
    }
}

File: workspace/1_CountMatchesInString/test/countmatches/CountMatchesInStringTest.java
package countmatches;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;

public class CountMatchesInStringTest {
	
	@Test
	public void noMatchesInNullString() {
		assertEquals(0, StringUtil.countMatches(null, 'b'));
	}
	
	@Test
	public void noMatchesInEmptyString() {
		assertEquals(0, StringUtil.countMatches("", 'b'));
	}
	
	@Test
	public void noMatchesInNonEmptyString() {
		assertEquals(0, StringUtil.countMatches("this string has no matches", 'b'));
	}
	
	@Test
	public void oneMatchInNonEmptyString() {
		assertEquals(1, StringUtil.countMatches("this string boldly has matches", 'b'));
	}
	
	@Test
	public void multipleMatchesInNonEmptyString() {
		assertEquals(3, StringUtil.countMatches("this string contains b even more than one time b", 'b'));
	}

}
